src/Akka.Hosting/AkkaHostedService.cs

```diff
@@ -11,30 +14,46 @@ namespace Akka.Hosting
     internal sealed class AkkaHostedService : IHostedService
     {
         private ActorSystem _actorSystem;
+        private readonly IServiceProvider _serviceProvider;
         private readonly AkkaConfigurationBuilder _configurationBuilder;
         private readonly IHostApplicationLifetime _hostApplicationLifetime;
+        private readonly ILogger<AkkaHostedService> _logger;
 
-        public AkkaHostedService(AkkaConfigurationBuilder configurationBuilder, IHostApplicationLifetime hostApplicationLifetime)
+        public AkkaHostedService(AkkaConfigurationBuilder configurationBuilder, IServiceProvider serviceProvider, ILogger<AkkaHostedService> logger)
         {
             _configurationBuilder = configurationBuilder;
-            _hostApplicationLifetime = hostApplicationLifetime;
+            _hostApplicationLifetime = serviceProvider.GetRequiredService<IHostApplicationLifetime>();
```

Comment:
I actually should change this back - came up during refactoring while we were working out instantiation issues.